src/core/thread/network_diagnostic.cpp

```diff
@@ -500,7 +508,6 @@ void NetworkDiagnostic::HandleDiagnosticGetQuery(Coap::Message &aMessage, const
         SuccessOrExit(error = message->SetPayloadMarker());
     }
 
-    messageInfo.SetSockAddr(Get<Mle::MleRouter>().GetMeshLocal16());
```

Comment:
do we need set source address here?

Reply:
Previous thought is that for the multicast query, the source should either be RLOC or LL64, and the answer source would be EID or LL64 according to existing select rules. It should be fine to select EID in the anwer.

Agree that it would be better to update it as well.

Reply:
agree it should be updated as well.